src/common/wal/src/config/kafka/metasrv.rs

```diff
@@ -12,50 +12,34 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-use std::time::Duration;
-
 use serde::{Deserialize, Serialize};
 
-use crate::config::kafka::common::{backoff_prefix, BackoffConfig};
-use crate::{TopicSelectorType, BROKER_ENDPOINT, TOPIC_NAME_PREFIX};
+use crate::config::kafka::common::{
+    backoff_prefix, kafka_topic_prefix, BackoffConfig, KafkaTopicConfig,
+};
+use crate::BROKER_ENDPOINT;
 
 /// Kafka wal configurations for metasrv.
 #[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
 #[serde(default)]
 pub struct MetasrvKafkaConfig {
     /// The broker endpoints of the Kafka cluster.
     pub broker_endpoints: Vec<String>,
-    /// The number of topics to be created upon start.
-    pub num_topics: usize,
-    /// The type of the topic selector with which to select a topic for a region.
-    pub selector_type: TopicSelectorType,
-    /// Topic name prefix.
-    pub topic_name_prefix: String,
-    /// The number of partitions per topic.
-    pub num_partitions: i32,
-    /// The replication factor of each topic.
-    pub replication_factor: i16,
-    /// The timeout of topic creation.
-    #[serde(with = "humantime_serde")]
-    pub create_topic_timeout: Duration,
     /// The backoff config.
     #[serde(flatten, with = "backoff_prefix")]
     pub backoff: BackoffConfig,
+    /// The kafka config.
+    #[serde(flatten, with = "kafka_topic_prefix")]
```

Comment:
To avoid introducing breaking changes.
```suggestion
    #[serde(flatten)]
```